backend/src/receipts.js

```diff
@@ -21,46 +22,57 @@ async function getIndexerCompatibilityReceiptActionKinds() {
   return INDEXER_COMPATIBILITY_RECEIPT_ACTION_KINDS;
 }
 
-async function getReceiptsList(blockHash) {
-  const receiptActions = await queryReceiptsList(blockHash);
+async function generateProperReceiptsList(receiptsList) {
```

Comment:
"proper" is subjective and not descriptive. I suggest `groupReceiptActionsIntoReceipts` and the input argument should be `receiptActions` (while the return value will be a list of receipts)